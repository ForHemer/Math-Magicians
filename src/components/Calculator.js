import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <section className="calculator-container">
        <div className="display">0</div>
        <div className="list-btn">
          <button className="list-btn-item" type="button">AC</button>
          <button className="list-btn-item" type="button">+/-</button>
          <button className="list-btn-item" type="button">%</button>
          <button className="list-btn-item operator" type="button">รท</button>
        </div>
        <div className="list-btn">
          <button className="list-btn-item" type="button">7</button>
          <button className="list-btn-item" type="button">8</button>
          <button className="list-btn-item" type="button">9</button>
          <button className="list-btn-item operator" type="button">x</button>
        </div>
        <div className="list-btn">
          <button className="list-btn-item" type="button">4</button>
          <button className="list-btn-item" type="button">5</button>
          <button className="list-btn-item" type="button">6</button>
          <button className="list-btn-item operator" type="button">-</button>
        </div>
        <div className="list-btn">
          <button className="list-btn-item" type="button">1</button>
          <button className="list-btn-item" type="button">2</button>
          <button className="list-btn-item" type="button">3</button>
          <button className="list-btn-item operator" type="button">+</button>
        </div>
        <div className="list-btn last-row">
          <button className="list-btn-item" type="button">0</button>
          <button className="list-btn-item" type="button">.</button>
          <button className="list-btn-item operator" type="button">=</button>
        </div>
      </section>
    );
  }
}

export default Calculator;
