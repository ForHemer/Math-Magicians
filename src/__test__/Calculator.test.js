import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('renders correctly', () => {
  const calc = renderer.create(<Calculator />).toJSON();
  expect(calc).toMatchSnapshot();
});
