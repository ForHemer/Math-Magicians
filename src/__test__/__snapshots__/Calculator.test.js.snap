// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="PageContainer"
>
  <div
    className="content"
  >
    <p>
      Let's do some math!
    </p>
  </div>
  <section
    className="calculator-container"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="list-btn"
    >
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="list-btn-item operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="list-btn"
    >
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="list-btn-item operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="list-btn"
    >
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="list-btn-item operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="list-btn"
    >
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="list-btn-item operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="list-btn last-row"
    >
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="list-btn-item"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="list-btn-item operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </section>
</main>
`;
